Improve auth flow

niccoloMG opened the pull request:
## Description

Improve auth flow by adding a loading component to the auth store state. Remove the dashboard layout and render the sign out button conditionally on main layout. 

## Task 
[Notion Task Link](https://www.notion.so/Tasks-538c1c6dbb2f4cba9a11c204ff418786?p=aa3f1d67267649119d77b25d182ccee8&pm=s)

## Screenshots (optional)

niccoloMG committed:
Add a loading state to auth store

Loading is by default true. Once we get the user info (and later,
user data as well) loading state wii be set to false to render
actual components.

niccoloMG committed:
update auth store if user exists after the auth state change

niccoloMG committed:
remove user info from auth store if user does not exist after the auth state change

niccoloMG committed:
Render a header in the main layout if there is user info authStore

niccoloMG committed:
remove the header text and add a signout button in the main layout

niccoloMG committed:
Delete dashboard layout as it is not needed anymore for signout

niccoloMG committed:
Render dashboard comp only if loading is false

berk245 commented on src/routes/+layout.svelte:
```diff
@@ -27,4 +42,7 @@
 </script>
 
+{#if $authStore.user}
+	<button on:click={authHandlers.signout}>Sign Out from main layout</button>
```
Let's change the button text here

niccoloMG replied:
Thank you, I missed that. Removed the 'from main layout' part

berk245 commented on src/routes/dashboard/+page.svelte:
```diff
@@ -1 +1,8 @@
+
+{#if !$authStore.loading}
+	<h1>User dashboard</h1>
+    <p>Welcome, user email: {$authStore.user}</p>
```
I will try to config vs code to run prettier on each save. Until then (or if you're using another ide), please make sure to do auto styling with prettier.

niccoloMG replied:
I don't know why code looks here like that. In my IDE, the indentation is correct. I ran the prettier again and pushed. Hoe it is fixed now

berk245 replied:
Interesting. Yes, now it's better

berk245 commented on src/routes/+layout.svelte:
```diff
@@ -2,14 +2,22 @@
 			currentPath = window.location.pathname;
 			if (!user && !noAuthRoutes.includes(currentPath)) {
 				// Unauthenticated user wants to go to a protected route
+				authStore.update((curr) => {
```
Moving the auth state updates outside of if blocks seems like a better idea to me, what do you think?
Here, we are not updating the state in the second block. There is also repetition. It'd be better to extract a function that takes user as parameter, and call it on each auth state change.

niccoloMG replied:
Thank you for the suggestion! I created a function and now call it on auth state change

berk245 replied:
Looks great! We might want to move the function to another file like helpers, but for now this will do great!

berk245 reviewed (changes requested):
Works well, good job! There are a few things that we can improve. See the comments for my suggestions. 

Another thing is related to commit messages. Please keep them short and if there is anything that requires an explanation, do so in the commit body. 
Two more nitpicks:
- Please try to keep your commits as atomic and accurate as possible (cea3596fb248781cf891c82e7f8fa404e689ea08, this commit does more than the message says)
- Please keep the capitalization consistent. Always start the subject line with a capital letter

niccoloMG committed:
Update sign out button text

niccoloMG committed:
Render dashboard if user is defined

Previously, we were only checking for loading state. That resulted
in momentary renders of dashboard before auto redirect to /login.
With this check, the render does not happen anymore and the jumpy
behavior is fixed.

niccoloMG committed:
Create an auth store update method and call it on auth state change

niccoloMG committed:
Improve code formatting

berk245 reviewed (approved):
Looks great!

tonyjojo reviewed (approved):
Thanks for your first contribution Nicco!
